instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/HttpClientTracer.java

```diff
@@ -183,7 +183,17 @@ private void setUrl(AttributeSetter setter, REQUEST request) {
       URI url = url(request);
       if (url != null) {
         netPeerAttributes.setNetPeer(setter, url.getHost(), null, url.getPort());
-        setter.setAttribute(SemanticAttributes.HTTP_URL, url.toString());
+        setter.setAttribute(
+            SemanticAttributes.HTTP_URL,
+            new URI(
+                    url.getScheme(),
+                    null,
+                    url.getHost(),
+                    url.getPort(),
+                    url.getPath(),
+                    url.getQuery(),
+                    url.getFragment())
+                .toString());
```

Comment:
hi @Hangzhi! I think it would be a worthwhile optimization here to only create a new URI object when the existing URI has user-info

Reply:
@Hangzhi, are you willing to address this comment?

Reply:
Thanks for the help @Hangzhi I've gone ahead and applied this suggestion to the PR